File: CS50x solutions/week_1/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);
    // for each row
    for (int row = 0; row < height; row++)
    {
        // for each collum
        for (int collum = 0; collum < height + 10; collum++)
        {
            // for space between hashes
            if (collum == height || collum == height + 1)
            {
                printf(" ");
            }
            // for the left side of pyramid
            else if (collum < height)
            {
                if (collum + row >= height - 1)
                {
                    printf("#");
                }
                else
                {
                    printf(" ");
                }
             }
             // for the right side of pyramid
             else if (collum > height)
             {
                if (collum - row < height + 3)
                {
                    printf("#");
                }
             }
        }
        printf("\n");
    }
}

File: CS50x solutions/week_6/sentimental-mario-more/mario.py
# TODO
from cs50 import get_int
height_list = [1, 2, 3, 4, 5, 6, 7, 8]

while 1:
    height = get_int("Height: ")
    if height in height_list:
        break
for row in range(height):
    for collum in range(height + 10):
        if collum == height or collum == height + 1:
            print(" ", end = '')
        elif collum < height:
            if collum + row >= height - 1:
                print("#", end = '')
            else:
                print(" ", end = '')
        elif collum > height:
            if collum - row < height + 3:
                print("#", end = '')
    print()